### DISCUSSION TITLE
Trailing dots in domains

### DISCUSSION THREAD

**Discussion by jflabelle81** [mdn/mdn-community#343] [Content]
I had a funny situation after a client made a press release with a typo in the URL with a period at the end of the domain. Technically I think it shouldn't work due to how DNS works (even though the trailing dot is a required DNS zone file format but not in use), browsers let it through, pointing to the DNS entry without the dot. (without removing the dot).

**I think the browser should automatically remove the dot or should not pass the request (invalid).**

The problem that this generated is that there are certain services that do checks at the domain level so the site was displaying badly due to the fact that certain checks were blocking the display. Then it's like a technical blur; Cloudflare is not able to manage this, neither are the other services... 

**In the end I think it should rather be managed on the browser side. What do you think?**

**Comment by bsmth** [comment]
Thanks for asking - that's a good question. Regarding FQDNs, there's some more information about this topic here: https://en.wikipedia.org/wiki/Fully_qualified_domain_name

If you're interested in a good read, Julia Evans has a good blog post about it I can recommend: https://jvns.ca/blog/2022/09/12/why-do-domain-names-end-with-a-dot-/

The likelihood of issues is with TLS expecting exact matches, but a lot of browsers handle this, see https://serverfault.com/questions/470719/https-and-trailing-dot-in-domain

I'm closing this as resolved, but feel free to reply if you think that's a mistake. You're also welcome to join our [Discord](https://developer.mozilla.org/discord) if you'd prefer to chat with us more about it! 

Thank you :)
